yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -544,6 +545,7 @@ private[spark] class Client(
       logInfo(s"Credentials file set to: $credentialsFile")
       val renewalInterval = getTokenRenewalInterval(stagingDirPath)
       sparkConf.set("spark.yarn.token.renewal.interval", renewalInterval.toString)
+      SparkHadoopUtil.get.startExecutorDelegationTokenRenewer(sparkConf)
```

Comment:
This code had change the configuration `spark.yarn.credentials.file`, and this configuration would be used in `DelegationTokenUpdate`, so it had to put the `start` after this.